pkg/operation/botanist/controlplane/kubeapiserver/deployment/networkpolicy.go

```diff
@@ -0,0 +1,300 @@
+				PodSelector: &metav1.LabelSelector{
+					MatchLabels: getAPIServerPodLabels(),
+				},
+			})
+
+		}
+
+		// in this case the konnectivity server is a separate deployment
+		// Allow connections from the APIServer pod to the konnectivity-server pod
+		// on its TLS port
+		if k.konnectivityTunnelEnabled && k.sniValues.SNIEnabled {
+			konnectivitiyServerHTTPSPort := intstr.FromInt(int(konnectivity.ServerHTTPSPort))
+			netpolAllowKubeApiserver.Spec.Egress = append(netpolAllowKubeApiserver.Spec.Egress, networkingv1.NetworkPolicyEgressRule{
+				To: []networkingv1.NetworkPolicyPeer{
+					{
+						PodSelector: &metav1.LabelSelector{
+							MatchLabels: konnectivity.GetLabels(),
+						},
+					},
+				},
+				Ports: []networkingv1.NetworkPolicyPort{
+					{
+						Protocol: &tcp,
+						Port:     &konnectivitiyServerHTTPSPort,
+					},
+				},
+			})
+		}
+
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *kubeAPIServer) deployAllowToShootAPIServer(ctx context.Context, netpolAllowToShootApiserver *networkingv1.NetworkPolicy) error {
+	tcp := corev1.ProtocolTCP
+	port443 := intstr.FromInt(443)
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.seedClient.Client(), netpolAllowToShootApiserver, func() error {
+		netpolAllowToShootApiserver.Annotations = map[string]string{
+			v1beta1constants.GardenerDescription: `|
+Allows Egress from pods labeled with 'networking.gardener.cloud/to-shoot-apiserver=allowed'
+to talk to Shoot's Kubernetes API Server.
+`,
+		}
+
+		netpolAllowToShootApiserver.Spec = networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					v1beta1constants.LabelNetworkPolicyToShootAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+				},
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{
+					To: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: getAPIServerPodLabels(),
+							},
+						},
+					},
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: &tcp,
+							Port:     &port443,
+						},
+					},
+				},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *kubeAPIServer) deployAllowFromShootAPIServer(ctx context.Context, netpolAllowFromShootApiserver *networkingv1.NetworkPolicy) error {
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.seedClient.Client(), netpolAllowFromShootApiserver, func() error {
+		netpolAllowFromShootApiserver.Annotations = map[string]string{
+			v1beta1constants.GardenerDescription: `|
+Allows Egress from Shoot's Kubernetes API Server to talk to pods labeled
+with 'networking.gardener.cloud/from-shoot-apiserver=allowed'.
```

Comment:
Can you use the constant here instead of "networking.gardener.cloud/from-shoot-apiserver"?

same above for other netpols